nutanix/resource_nutanix_virtual_machine.go

```diff
@@ -1585,6 +1590,24 @@ func preFillResUpdateRequest(res *v3.VMResources, response *v3.VMIntentResponse)
 	}
 	res.NicList = nold
 
+	sold := make([]*v3.VMDisk, len(response.Status.Resources.DiskList))
```

Comment:
You could keep this and return the fields ForceNew and the rest let them ForceNew as they were before, I think shouldn't delete the DiskList.